.github/workflows/publish_main.yaml

```
@@ -64,25 +64,43 @@ jobs:
         run: |
           set +o pipefail
           cd demos/${{ matrix.demo_directory }}
-          export OTEL_METRICS_EXPORTER=console
-          export OTEL_LOGS_EXPORTER=console
-          export OTEL_TRACES_EXPORTER=console
-          export OTEL_SHELL_SDK_OUTPUT_REDIRECT="./otlp"
           otlp="$(mktemp)"
           stdout="$(mktemp)"
           stderr="$(mktemp)"
-          mkfifo "$OTEL_SHELL_SDK_OUTPUT_REDIRECT"
-          while true; do cat "$OTEL_SHELL_SDK_OUTPUT_REDIRECT" >> "$otlp"; done &
-          pid="$!"
+          echo "
+          receivers:
+            otlp:
+              protocols:
+                grpc:
+                  endpoint: 0.0.0.0:4317
+                http:
+                  endpoint: 0.0.0.0:4318
+          processors:
+            batch:
+          exporters:
+            file:
+              path: /etc/otelcol/otlp.json
+          service:
+            pipelines:
+              traces:
+                receivers: [otlp]
+                exporters: [file]
+                processors: [batch]
+          " > config.yaml
+          touch otlp.json && chmod 0666 otlp.json
+          sudo docker create --rm --mount type=bind,source=$(pwd)/config.yaml,target=/etc/otelcol/config.yaml --mount type=bind,source=$(pwd)/otlp.json,target=/etc/otelcol/otlp.json --network=host --name=collector otel/opentelemetry-collector:latest
+          sudo docker start collector
+          sleep 10
           cat meta.json | jq -r .setup | (grep -v '^null$' || true) | sh
           bash -e demo.sh 1> "$stdout" 2> "$stderr" || true
-          kill -9 "$pid"
-          wait
-          sleep 10
+          sleep 15
+          sudo docker stop collector
```

```suggestion
          sudo docker rm -f collector
```
Nit: Since we're also removing `config.yaml` and `otlp.json` I think we should also scrape the container. It shouldn't be a problem to rm -f it because the container is not needed afterwards.

---

the container is scraped by the `--rm` flag in its create command. removing config.yaml and otlp.json is more important because if we dont remove them, they will be committed with the rest